### PR TITLE
Kubelet should wait for cloud-init to ensure a fully configured node

### PR DESCRIPTION
**Pull request by PerGon**
*Issue https://github.com/awslabs/amazon-eks-ami/issues/647*

*Description of changes:*
This will make the kubelet service to wait for cloud-init before starting. According to [these docs](https://cloudinit.readthedocs.io/en/latest/topics/boot.html#network) (and some investigation), cloud-init is responsible to initialise disks and mounts. Waiting for it before starting the kubelet, will ensure the node will be "ready".

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by heybronson — APPROVED** [review]

**Comment by PerGon** [comment]
@heybronson thank you for the approval.
Is there anything else I need to do, or is it just a matter of time until the maintainers merge this PR?

**Pull request closed without merging**
